tests/token_token_create_transaction.rs

```diff
@@ -0,0 +1,306 @@
+    let token_create_response = match token_create_tx.max_transaction_fee(Hbar::new(100)).execute(client).await {
+        Ok(tx_response) => tx_response,
+        Err(e) => panic!("Token Create Transaction failed with error: {}", e)
+    };
+
+    // Get transaction receipt
+    let receipt = match token_create_response.get_receipt(client).await {
+        Ok(receipt) => receipt,
+        Err(e) => panic!("Transaction Receipt failed with error: {}", e)
+    };
+
+    // Ectract and return only token id
+    match receipt.token_id {
+        Some(token_id) => return token_id,
+        None => panic!("Token Id retrieval failed")
+    };
+}
+
+// Fetch token data on chain with TokenInfo query
+async fn get_token_info(client: &Client, token_id: TokenId) -> TokenInfo{
+    let _ = match TokenInfoQuery::new()
+        .token_id(token_id)
+        .execute(client)
+        .await {
+            Ok(token_info) => return token_info,
+            Err(e) => panic!("Token Info Query failed with error: {}", e)
+        };
+}
+
+// Compare all data from the token onchain (token_info) with the previously used data (used_data)
+fn check_token_info(token_info: TokenInfo, used_data: &UsedData) {
+    assert_eq!(token_info.name, used_data.name, "On chain name does not match name from used_Data.");
+    assert_eq!(token_info.token_type, used_data.token_type, "On chain token_type does not match token_type from used_Data.");
+    assert_eq!(token_info.symbol, used_data.symbol, "On chain symbol does not match symbol from used_Data.");
+    assert_eq!(token_info.decimals, used_data.decimals, "On chain decimals do not match decimals from used_Data.");
+    assert_eq!(token_info.total_supply, used_data.initial_supply, "On chain total_supply does not match initial_supply from used_Data.");
+    assert_eq!(token_info.treasury_account_id, used_data.account_id, "On chain treasury_account_id does not match account_id from used_Data.");
+    assert_eq!(token_info.default_freeze_status.unwrap(), used_data.default_freeze_status, "On chain default_freeze_status does not match default_freeze_status from used_Data.");
+    assert_eq!(token_info.auto_renew_account.unwrap(), used_data.account_id, "On chain auto_renew_account does not match account_id from used_Data.");
+    assert_eq!(token_info.auto_renew_period.unwrap(), used_data.auto_renew_period, "On chain auto_renew_period does not match auto_renew_period from used_Data.");
+    assert_eq!(token_info.token_memo, used_data.memo, "On chain token_memo does not match memo from used_Data.");
+    assert_eq!(token_info.custom_fees, used_data.custom_fees, "On chain custom_fees do not match custom_fees from used_Data.");
+    
+    match &used_data.public_key {
+        PublicKeyType::PublicKey(key) => {
+            let key = Key::from(key.clone());
+            assert_eq!(token_info.admin_key.unwrap(), key, "On chain admin_key does not match public_key from used_Data.");
```

Comment:
(substitute this for the ones below as well)

`assert_eq!(token_info.admin_key, Some(key))`